pkg/ipcache/ipcache.go

```diff
@@ -497,6 +497,15 @@ func (ipc *IPCache) RemoveMetadata(prefix netip.Prefix, resource ipcacheTypes.Re
 	ipc.TriggerLabelInjection()
 }
 
+// RemoveAllMetadata removes all metadata associated with this resource
+func (ipc *IPCache) RemoveAllMetadata(prefix netip.Prefix, resource ipcacheTypes.ResourceID) {
```

Comment:
Could you explain a bit more deeply the rationale behind removing all? Does this mean that if two users of the new API insert metadata associations for different resources, then some other user comes along and uses this API, it will delete all of the metadata associations from the other user? That sounds counter-intuitive to the goals of having the ipcache itself receive desired state coming in and then handling the merging / resolution internally.

Reply:
I'm thinking about a use case like this:

- Policy engine creates an association that IP 192.0.2.3 needs a rich identity corresponding to a policy statement
- Endpoint watcher observes that 192.0.2.3 corresponds to kube-apiserver and adds an association corresponding to that resource
- Node manager observes that 192.0.2.3 is a remote node in the cluster.
- At some point 192.0.2.3 gets removed from the cluster, now `RemoveAllMetadata()` deletes the policy association so policy to that remote peer no longer works until... maybe the user restarts the agent?

Reply:
That's a fair question. I think my original intent was to keep the behavior as close to the previous code as possible (and the previous code issues a delete-all too). I think you make an excellent argument though, we should not delete everything here. I'll attempt to replace it with a regular remove labels.